src/select.scss

```diff
@@ -81,34 +96,56 @@ $block: #{$fd-namespace}-select;
     }
 
     &.is-readonly {
-      padding-right: $fd-select-padding-x;
-      padding-left: $fd-select-padding-x;
+      .#{$block}__text-content {
+        padding-right: $fd-select-padding-x;
+        padding-left: $fd-select-padding-x;
+      }
+
       cursor: text;
     }
   }
 
   &__text {
+    @include fd-reset();
+
     font-size: var(--sapFontSize);
   }
 
   &__text-content {
+    @include fd-reset();
+
+    padding-left: $fd-select-padding-x;
     min-width: 5rem;
+
+    @include fd-rtl() {
+      padding-left: 0;
+      padding-right: $fd-select-padding-x;
+    }
   }
 
   &--compact {
     .#{$block}__control {
       height: $fd-form-input-height--compact;
-      padding-left: $fd-select-padding-compact-x;
+
+      .#{$block}.#{$block}__text-content {
+        padding-left: $fd-select-padding-compact-x;
+      }
 
       @include fd-rtl() {
         text-align: right;
-        padding-right: $fd-select-padding-compact-x;
         padding-left: 0;
+
+        .#{$block}.#{$block}__text-content {
```

Comment:
I didn't find a usage of `fd-select fd-select__text-content`

Reply:
true i fixed it to just .#{$block}__text-content